apps/process-new-granules/src/process_new_granules.py

```diff
@@ -1,83 +1,31 @@
-from copy import deepcopy
+import os
 from datetime import datetime, timedelta, timezone
 
-import asf_search
-import dateutil.parser
+import boto3
 
 import dynamo
 
 
-def get_unprocessed_granules(subscription):
-    processed_jobs, _ = dynamo.jobs.query_jobs(
-        user=subscription['user_id'],
-        name=subscription['job_specification']['name'],
-        job_type=subscription['job_specification']['job_type'],
+def handle_subscription_batch(subscriptions: list[dict], cutoff_date: datetime, worker_function_arn: str) -> None:
+    payload = {
+        'subscriptions': subscriptions,
+        'cutoff_date': cutoff_date.isoformat(),
+    }
+    client = boto3.client('lambda')
```

Comment:
Err on the side of declaring boto3 clients in global space, so many api calls (and subsequent lambda executions) can share the same http session:

https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html
> Take advantage of execution environment reuse to improve the performance of your function. Initialize SDK clients and database connections outside of the function handler, and cache static assets locally in the /tmp directory. Subsequent invocations processed by the same instance of your function can reuse these resources. This saves cost by reducing function run time.
> 
> To avoid potential data leaks across invocations, don’t use the execution environment to store user data, events, or other information with security implications. If your function relies on a mutable state that can’t be stored in memory within the handler, consider creating a separate function or separate versions of a function for each user.